docs/usage/constraints.md

```diff
@@ -0,0 +1,32 @@
+# Adding a Constraint to a table
+
+Check constraints are a way to enforce that only data that meets the constraint is allowed to be added to the table.
+
+## Add the Constraint
+
+```python
```

Comment:
Any chance you want to move new format, e.g. a quick example:
1. https://github.com/delta-io/delta-rs/blob/main/docs/usage/index.md
2. referring to function under in https://github.com/delta-io/delta-rs/blob/main/docs/src/python/delta_table.py and https://github.com/delta-io/delta-rs/blob/main/docs/src/rust/delta_table.rs
3. in addition, if there is something in the API for more info then you can add those links here and refer in the example in md file: https://github.com/delta-io/delta-rs/blob/main/docs/_build/links.yml
4.